swap_meet/vendor.py

```diff
@@ -1,2 +1,67 @@
+from swap_meet import item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = inventory or []
+    
+    def add(self, one_item):
+        if not one_item in self.inventory:
+            self.inventory.append(one_item)
+        return one_item
+    
+    def remove(self, one_item):
+        if not one_item in self.inventory:
+            return False
+        elif one_item in self.inventory:
+            self.inventory.remove(one_item)
+            return one_item
+    
+    def get_by_id(self, id):
+        for item in self.inventory:   
+            if item.id == id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        for item in self.inventory:
+            if my_item in self.inventory and their_item in other_vendor.inventory:                
+                self.inventory.remove(my_item)
+                other_vendor.inventory.append(my_item)
+                other_vendor.inventory.remove(their_item)
+                self.inventory.append(their_item)
+                return True
+        return False     
+    
+    def swap_first_item(self, other_vendor):
+        if self.inventory == [] or other_vendor.inventory == []:
+            return False
```

Comment:
Similar to my comment above, this function can improve if we utilize other instance methods that already exist. Consider this code:

```python
        self.add(their_first_item)
        self.remove(first_item)
        other_vendor.add(first_item)
        other_vendor.remove(their_first_item)
```

and if we push it even farther, we can actually use `swap_items` to make this even more elegant:

```python
        self.swap_items(other_vendor, first_item, their_first_item)
```